### PR TITLE
Enable FireSim distributed elaboration

### PR DESCRIPTION
**Pull request by timsnyder-siv**
**Related issue**: firesim/firesim#646

**Type of change**: other enhancement

**Impact**: rtl change | software change | unknown | other

**Release Notes**
* build.sbt assembly merge strategy made more conservative and custom strategy for preferring non-rocketchip artifacts
* `chipyard.config.WithBootROM` now looks for contentsFile in `TargetDirKey` instead of relative to current-working directory
   * removes `firesim.firesim.WithBootROM` because things are simplified

### CODE REVIEW

**Comment by timsnyder-siv** [comment]
Looks like I will need to merge forward on dev but I wanted to go ahead and post this so that @davidbiancolin and others could see it.

This work requires chipsalliance/rocket-chip#2725 so that `TargetDirKey` is in the Config.

**Review by abejgonzalez — APPROVED** [review]
Everything minus the `build.sbt` stuff looks good to me (the `build.sbt` stuff is a bit out of my comfort zone). Its nice that CY/FireSim can share the same `WithBootROM` frag.

**Comment by davidbiancolin** [comment]
SBT side things look fine to me.

**Comment by timsnyder-siv** [comment]
I'm working on merging local-chisel34 branch into this because @davidbiancolin wanted to wait to merge this until after the chisel bump.

**Comment by timsnyder-siv** [comment]
Since I merged `local-chisel34` into my topic branch before it was merged to `dev`, I now have all kinds of changes in this PR that have nothing to do with the original topic.

I'm going to try cleaning this up by force-pushing my target branch and merging with `dev` now that `local-chisel34` is merged.

**Comment by timsnyder-siv** [comment]
@davidbiancolin I'm seeing that CY is currently pointing to https://github.com/chipsalliance/rocket-chip/tree/577994e38e3115cafa3a232b0fc60584aacb996e and that doesn't have chipsalliance/rocket-chip#2725 that this requires.

Are you guys planning to bump RC before the release?

Here's the comparison showing what would have to be bumped to enable this PR https://github.com/chipsalliance/rocket-chip/compare/577994e38e3115cafa3a232b0fc60584aacb996e...0049f6ee08e88c6736800789ea7a90e8555d50f1

**Comment by davidbiancolin** [comment]
Oops... We definitely should've bumped further along in RC, i neglected to check. I'll try to get around to that next week, otherwise i'm open to a PR.

**Comment by timsnyder-siv** [comment]
> Oops... We definitely should've bumped further along in RC, i neglected to check. I'll try to get around to that next week, otherwise i'm open to a PR.

opened #742

**Comment by timsnyder-siv** [comment]
hmm.  Does anyone know why the checks aren't running after I rebased and force-pushed this?  I put the 'DO NOT MERGE' label on this PR needs to merge after #742 (or someone needs to say it's okay for rocket to not be on master).

**Comment by jerryz123** [comment]
The checks don't run on PRs from forks. @abejgonzalez do you remember why that was the case?
Also @abejgonzalez @davidbiancolin  can we just merge these in without waiting for the checks? The changes seem to be mostly innocuous.

**Comment by timsnyder-siv** [comment]
I'm also happy to resubmit with a local branch now that I have write perms.

**Comment by timsnyder-siv** [comment]
I'm rebasing this onto dev and opening a local branch PR that will replace this one.  Don't see a way to change the remote source branch of a PR so I'm closing this one.

**Pull request closed without merging**
